qualtran/bloqs/for_testing/random_select_and_prepare.py

```diff
@@ -0,0 +1,157 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from functools import cached_property
+from typing import Optional, Sequence, Tuple
+
+import attrs
+import cirq
+import numpy as np
+from attrs import frozen
+from numpy.typing import NDArray
+
+from qualtran import BloqBuilder, BoundedQUInt, QBit, Register, SoquetT
+from qualtran.bloqs.for_testing.random_gate import RandomGate
+from qualtran.bloqs.qubitization_walk_operator import QubitizationWalkOperator
+from qualtran.bloqs.select_and_prepare import PrepareOracle, SelectOracle
+
+
+@frozen
+class RandomPrepareOracle(PrepareOracle):
+    U: RandomGate
```

Comment:
Add a docstring to the Bloq. Should we add a post init check to verify that first column of the matrix is all real numbers?

Reply:
yep! added a post init check